### PR TITLE
Reorganize mem safety

### PR DESCRIPTION
**Pull request by nano-o**

### CODE REVIEW

File: proof/ec_opts.saw [hunk 1]
```diff
@@ -204,13 +211,14 @@ POINTonE2_add_ov <- verify "POINTonE2_add" assembly_overrides POINTonE2_add_spec
...
-POINTonE2_add_affine_ov <- crucible_llvm_verify m "POINTonE2_add_affine" assembly_overrides false POINTonE2_add_affine_spec abc;
+POINTonE2_add_affine_ov <- verify "POINTonE2_add_affine" assembly_overrides POINTonE2_add_affine_spec;
 POINTonE2_add_affine_alias_ov <- verify "POINTonE2_add_affine" assembly_overrides POINTonE2_add_affine_alias_spec;
 POINTonE1_double_ov <- verify "POINTonE1_double" assembly_overrides POINTonE1_double_spec;
 POINTonE1_double_alias_ov <- verify "POINTonE1_double" assembly_overrides POINTonE1_double_alias_spec;
 POINTonE2_double_ov <- verify "POINTonE2_double" assembly_overrides POINTonE2_double_spec;
 POINTonE2_double_alias_ov <- verify "POINTonE2_double" assembly_overrides POINTonE2_double_alias_spec;
 POINTonE1_is_equal_ov <- verify "POINTonE1_is_equal" assembly_overrides POINTonE1_is_equal_spec;
 POINTonE2_is_equal_ov <- verify "POINTonE2_is_equal" assembly_overrides POINTonE2_is_equal_spec;
+//POINTonE1_mult_w5_ov <- really_verify "POINTonE1_mult_w5" assembly_overrides (POINTonE1_mult_w5_spec 636); // TODO: how many bits?
```
Comment (andreistefanescu) [inline proof/ec_opts.saw:222]: is this used?

File: proof/hash_to_field.saw [hunk 2]
```diff
@@ -44,19 +32,14 @@ let hash_to_field_spec nelems aug_len msg_len DST_len = do {
...
 
-let verify_expand_message_for len_in_bytes aug_len msg_len DST_len = do {
-  block_data_order_ovs <- make_block_data_order_ovs [aug_len, msg_len, eval_size {| DST_len+4 |}];
-  r <- crucible_llvm_verify m "expand_message_xmd" (concat bcopy_ovs (concat block_data_order_ovs [blst_sha256_emit_ov, blst_sha256_hcopy_ov])) false (expand_message_spec len_in_bytes aug_len msg_len DST_len) abc;
-  return r;
+let verify_hash_to_field_for aug_len msg_len DST_len nelems = do {
+  let L = 64; // sizeof(vec384) + 128/8 = 48 + 16
+  let len_in_bytes = eval_size {| nelems * L |};
+  block_data_order_ovs <- make_block_data_order_ovs [aug_len, msg_len, eval_size {| (33+DST_len+1+9)+63 |}]; // last size is b_i_blocks from hash_to_field.c
+  //expand_message_xmd_ov <- verify "expand_message_xmd" (concat bcopy_ovs (concat block_data_order_ovs [blst_sha256_emit_ov, blst_sha256_hcopy_ov])) (expand_message_spec len_in_bytes aug_len msg_len DST_len); // TODO: does not terminate
```
Comment (andreistefanescu) [inline proof/hash_to_field.saw:39]: what is causing this change?

**Review by andreistefanescu — APPROVED** [review]
Looks good, see a couple of inline comments

File: proof/ec_opts.saw (hunk 1 above)
Comment (nano-o) [reply proof/ec_opts.saw]: fixed: it's now verified for the right number of bits

File: proof/hash_to_field.saw (hunk 2 above)
Comment (nano-o) [reply proof/hash_to_field.saw]: I don't know. It seems that Z3 chokes on it, but abc does fine...

**Review by andreistefanescu — APPROVED** [review]
Looks good, thank you @nano-o

**Pull request merged**